crates/nostr/src/event/builder.rs

```diff
@@ -712,6 +712,43 @@ impl EventBuilder {
         Self::new(Kind::EventDeletion, reason.into()).tags(tags)
     }
 
+    /// Request to vanish with reason
+    ///
+    /// Must include at least one relay.
+    ///
+    /// <https://github.com/nostr-protocol/nips/blob/master/62.md>
+    pub fn request_vanish_with_reason<I, S>(relays: I, reason: S) -> Result<Self, Error>
```

Comment:
Here we can take an enum to allow to request a global vanish. Or, like you suggested, request a global vanish if no relay is specified.

In the case of enum may be:

```rust
pub enum VanishTarget {
    Global,
    Relay(RelayUrl)
}
```

I know that spec allow to specify many relays, but I think that if many relays are specified, the aim is probably to delete the events from all the relays, so the `ALL_RELAYS` is faster and easier. If we choose the enum way, it can be added in a `nip62` module.

Reply:
> the aim is probably to delete the events from all the relays

I don't think so, maybe I want only to delete my events from 2~3 relays.

I don't think there is any issue to make it `Vec<RelayUrl>` and make a helper functions, one for one relay, many relays and all relays (GLOBAL)

Reply:
Ok, so something like this:

```rust
pub enum VanishTarget {
    Global,
    Relays(Vec<RelayUrl>)
}

impl VanishTarget {
    pub fn global() -> Self { ... }

   pub fn relay(relay: RelayUrl) -> Self { ... }

   pub fn relays<I>(relays: I) -> Self 
   where
      I: IntoIterator<Item = RelayUrl>
   { ... }
}
```